### ISSUE TITLE
az webapp deployment source config-zip command often fail and always log warnings.

### ISSUE THREAD

**Issue by phuot** [Azure/azure-cli#12125]
Az Cli version: 2.0.77

**Describe the bug**
When trying to use az webapp deployment we often have error and the deployment does not always complete.
It happen most of the time when deploying to a slot that has been created few seconds ago.
_more details in step to reproduce section_

**To Reproduce**
here is my creation workflow (in case other steps might be causing the issue):
```
az webapp deployment slot create --name $webAppName --resource-group $resourceGroupName --slot $slotName --configuration-source $webAppName

az webapp identity assign --resource-group $resourceGroupName --name $webAppName --slot $slotName --query principalId --output tsv

az webapp config appsettings set --resource-group $resourceGroupName --name $webAppName --slot $slotName --settings APPINSIGHTS_INSTRUMENTATIONKEY=$appInsightKey WEBSITE_LOAD_CERTIFICATES=*

az webapp deployment source config-zip --resource-group $resourceGroupName --name $webAppName --src $src --slot $slotName
```

sometime the error is similar to :
```
az : WARNING: Getting scm site credentials for zip deployment
At line:79 char:15
+ ...   $result = az webapp deployment source config-zip --resource-group $ ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (WARNING: Gettin... zip deployment:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
WARNING: Starting zip deployment. This operation can take a while to complete ...
WARNING: Deployment endpoint responses with status code 503
```

sometime we receive (replaced sensitive info with *****):
```
az : WARNING: Getting scm site credentials for zip deployment
At line:79 char:15
+ ...   $result = az webapp deployment source config-zip --resource-group $ ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (WARNING: Gettin... zip deployment:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
WARNING: Starting zip deployment. This operation can take a while to complete ...
WARNING: Deployment endpoint responses with status code 202
WARNING: Configuring default logging for the app, if not already enabled
ERROR: Zip deployment failed. {'id': 'temp-8f69cfd5', 'status': 3, 'status_text': '', 'author_email': 'N/A', 'author': 'N/A', 'deployer': 'Push-Deployer', 'message': 'Deploying from pushed zip file', 'progress': '', 'received_time': 
'2020-02-10T19:12:51.4052348Z', 'start_time': '2020-02-10T19:12:51.4052348Z', 'end_time': '2020-02-10T19:12:58.7646024Z', 'last_success_end_time': None, 'complete': True, 'active': False, 'is_temp': True, 'is_readonly': False, 'url': 
'https://*********-test.scm.azurewebsites.net/api/deployments/latest', 'log_url': 'https://*********-test.scm.azurewebsites.net/api/deployments/latest/log', 'site_name': '*********', 'provisioningState': None}. 
Please run the command az webapp log tail
                           -n *********-g *********
```

If I look at the KUDU logs (https://*********-test.scm.azurewebsites.net/api/deployments/latest/log), I see the following error:
```
{
Message: "An error has occurred.",
ExceptionMessage: "No log found for 'latest'.",
ExceptionType: "System.IO.FileNotFoundException",
StackTrace: " at Kudu.Core.Deployment.DeploymentManager.GetLogEntries(String id) in C:\Kudu Files\Private\src\master\Kudu.Core\Deployment\DeploymentManager.cs:line 94 at Kudu.Services.Deployment.DeploymentController.GetLogEntry(String id) in C:\Kudu Files\Private\src\master\Kudu.Services\Deployment\DeploymentController.cs:line 375"
}
```

**Expected behavior**
I would expect that the Az CLI would not return unnecessary warning since it breaks my build pipeline (which does not expect any warning/error).
I would expect that the Az CLI would be able to do deployment without throwing errors.

**Environment summary**
azure-cli                         2.0.77
run on windows 10 or in azure pipeline.

**Comment by arrownj** [comment]
@phuot , thanks for reporting, we are looking at this.

**Comment by nikunjrc-azcli** [comment]
I am also facing same issue while deploying zip thru az cli....

**Comment by panchagnula** [comment]
az webapp deployment source config-zip has a default timeout so if the deployment takes longer, the command throws an error & exists. Run the command with a long timeout using --timeout

**Comment by yikma** [comment]
We are currently running into a similar issue. In regards to adding the --timeout parameter to the command, in the documentation found: https://docs.microsoft.com/en-us/cli/azure/webapp/deployment?view=azure-cli-latest, the --timeout parameter is not present however, it does exist as part of the az group deployment commands under the wait section. Are we still able to run -- timeout with az webapp deployment as a parameter.

**Comment by panchagnula** [comment]
closing due to lack of response.

**Comment by phuot** [comment]
sorry I missed the notification about the --timeout.
We tried and the --timeout does not solve the issue.

However I have a theory on why such issue occurs and I think I have found a workaround (for the issue where the command does not always complete).

I suspect that the following happens:

1. I deploy a new slot 
2. I apply some appSetting config change (which would be detected within 30 sec)
3. Then I starts the deployment
While the deployment is in progress, the web app detect that the AppSettings changed and the AppPool is recycled (slot is restarted). Hence the Az deployment lose its connection and get cancelled.

Workaround would be to add a sleep of a little more than 30 seconds after updating the configuration before executing the deployment step.

So far this workaround works.
